core/src/main/java/org/elasticsearch/index/mapper/TypeParsers.java

```diff
@@ -56,15 +56,30 @@
     private static final DeprecationLogger DEPRECATION_LOGGER = new DeprecationLogger(Loggers.getLogger(TypeParsers.class));
     private static final Set<String> BOOLEAN_STRINGS = new HashSet<>(Arrays.asList("true", "false"));
 
-    public static boolean nodeBooleanValue(String name, Object node, Mapper.TypeParser.ParserContext parserContext) {
-        // TODO: remove this leniency in 6.0
-        if (BOOLEAN_STRINGS.contains(node.toString()) == false) {
-            DEPRECATION_LOGGER.deprecated("Expected a boolean for property [{}] but got [{}]", name, node);
+    //TODO 22298: Remove this method and have all call-sites use <code>XContentMapValues.nodeBooleanValue(node)</code> directly.
+    public static boolean nodeBooleanValue(String fieldName, String propertyName, Object node, Mapper.TypeParser.ParserContext
+        parserContext) {
+        if (parserContext.indexVersionCreated().onOrAfter(Version.V_6_0_0_alpha1_UNRELEASED)) {
+            return XContentMapValues.nodeBooleanValue(node);
+        } else {
+            if (BOOLEAN_STRINGS.contains(node.toString()) == false) {
```

Comment:
No, not really. I think originally I just moved the existing method and just did not think about this aspect. But it's already corrected now as per your suggestion.